qiskit_nature/algorithms/pes_samplers/bopes_sampler.py

```diff
@@ -113,11 +125,26 @@ def sample(self, problem: BaseProblem, points: List[float]) -> BOPESSamplerResul
         self._driver = problem.driver
         # We have to force the creation of the solver so that we work on the same solver
         # instance before and after _gss.solve
-        self._gss.get_qubit_operators(problem, None)
-        if isinstance(self._gss.solver, VariationalAlgorithm):  # type: ignore
+        if isinstance(self._gss, ExcitedStatesSolver):
+            if hasattr(self._gss, "_gsc"):
+                self._gss._gsc.get_qubit_operators(problem, None)
+        else:
+            self._gss.get_qubit_operators(problem, None)
```

Comment:
If doing the above, this could be significantly simpilfied

Reply:
This was significantly simplified but for another reason:
The get_qubit_operators() was added in the abstract GroundStateSolver and ExcitedStatesSolver. Their implementation in QEOM, ExcitedStatesEigensolver and GroundStateEigensolver was provided (qeom calls GroundStateEigensolver.get_qubit_operators, ExcitedStatesEigensolver has its own which is almost a copy of the one of GroundStateEigensolver, GroundStateEigensolver has not changed).

It comes down to one line in the end:
`self._solver_wrapper.get_qubit_operators(problem=problem, aux_operators=None)`